doc/index.rst

```diff
@@ -57,6 +57,7 @@ Source code for the Zephyr Project is maintained in the Zephyr Project's
    subsystems/subsystems.rst
    contribute/contribute.rst
    security/security.rst
+   code_data_relocation/relocation.rst
```

Comment:
This doesn't seem to be the right place to put this document in the overall site organization.  I'd recommend this go into the developer guide (application/index.rst) and not the top level index.  This means also moving the relocation.rst file into the application folder.